apps/system/js/homescreen_launcher.js

```diff
@@ -216,7 +160,9 @@
         return null;
       }
       if (typeof this._instance == 'undefined') {
-        this._instance = new HomescreenWindow(this._currentManifestURL);
+        console.trace();
+        this._instance = new window.HomescreenWindow(this._currentManifestURL);
```

Comment:
nit: we already define ```HomescreenWindow``` as a global, so maybe we should just access it directly instead of through ```window```?